Improve vagrant setup reliability

MatthewWilkes opened the pull request:
## Description
This updates the vagrant setup to do the following things:

1. Pins the vm image to a modern version
2. Installs Python 3.6 and Postgres 10.6 rather than the default versions
3. Fixes a bug in autogenerating the global dallinger config
4. Switches the default Python for dallinger to 3.6
5. Works around a virtualbox bug where shared folders can get out of sync, which made tox runs fail

## Motivation and Context
Easier getting started path for new contributors

## How Has This Been Tested?
I've run the build process and then run the tests. I'll be testing more this week, so please don't merge immediately.

vlall commented:
Is this complete? I think the failed tests are my concern, but last time I tested it was working

vlall commented:
I have to change vbguest to another name i assume, any ideas?

```
Bringing machine 'default' up with 'virtualbox' provider...
There are errors in the configuration of this machine. Please fix
the following errors and try again:

Vagrant:
* Unknown configuration section 'vbguest'.
```

vlall commented on Vagrantfile:
```diff
@@ -3,6 +3,8 @@
 Vagrant.configure("2") do |config|
   config.vm.box = "ubuntu/xenial64"
+  config.vm.box_version = "= 20190204.3.0"
+  config.vbguest.auto_update = true
```
Here's the `vbguest` line that might be causing my error

vlall commented:
https://github.com/Dallinger/Dallinger/pull/1640/files#r261078501

vlall commented:
Two things i notice:
- Dallinger version is 5.05 instead of 5.06 (maybe this is a rebasing issue)
- the URL linked is 
```❯❯ Initial recruitment list:
http://10.0.2.15:5000/ad?recruiter=hotair&assignmentId=XPATQT&hitId=IDLX3Q&workerId=RBYFEX&mode=debug```
that URL fails to pull anything up, and if i try 
`http://localhost:5000/ad?recruiter=hotair&assignmentId=XPATQT&hitId=IDLX3Q&workerId=RBYFEX&mode=debug` that link does not work. 0.0.0.0 and 127.0.0.1 don't work either. So i can't access the experiment in debug mode.

MatthewWilkes commented:
@vlall I rebased to `origin/master`, it shows version 5.0.6 now. I also believe I've fixed the IP issue. Note, it will require the VM to be reprovisioned to fix this, so you should do:

    vagrant destroy
    vagrant up

vlall commented:
My IP issue and the requirements issue seem to be fixed
